agent/engine/task_manager.go

```diff
@@ -148,6 +149,22 @@ func (mtask *managedTask) overseeTask() {
 		if !mtask.GetKnownStatus().Terminal() {
 			// If we aren't terminal and we aren't steady state, we should be able to move some containers along
 			llog.Debug("Task not steady state or terminal; progressing it")
+
+			// TODO:
+			// 1) Determine better way to check feature enablement and
+			// cgroup hierarchy setup
+			// 2) Add new task resources provisioned state
+			if mtask.Task.CgroupEnabled() {
+				err := mtask.SetupCgroup()
+				if err != nil {
+					seelog.Errorf("Unable to set up task cgroup: %v", err)
+					// If task cgroup creation fails, set task status to stopped
+					// with valid reason
+					mtask.SetKnownStatus(api.TaskStopped)
```

Comment:
We should not be modifying the task's known status to 'STOPPED' without also modifying it's desired status (which we can safely presume is 'RUNNING' here)